src/test/java/com/intel/gkl/compression/DeflaterUnitTest.java

```diff
@@ -61,6 +68,120 @@ public void loadLibrary() {
+        Inflater intelInflater = new IntelInflaterFactory().makeInflater(nowrap);
+
+        //act
+        intelDeflater.setInput(input,0,input.length);
+        intelDeflater.finish();
+        intelDeflater.deflate(compressed,0,compressed.length);
+        intelDeflater.end();
+
+        intelInflater.setInput(compressed,0,compressed.length);
+        intelInflater.inflate(decompressed,0,decompressed.length);
+        intelInflater.end();
+
+        //assert
+        Assert.assertEquals(decompressed,input);
+    }
+
+    @Test(enabled = true, dataProviderClass = CompressionDataProviders.class, dataProvider = "compressionCases")
+    public void intelDeflationJavaInflationCompatibilityTest(int level, boolean nowrap) throws DataFormatException {
+        //arrange
+        final int len = TestingUtils.SIZE_4MB;
+        final int outputLen = (int) (1.2*TestingUtils.SIZE_4MB);
+
+        final byte[] input = TestingUtils.generateRandomDNAArray(len);
+        final byte[] compressed = new byte[outputLen];
+        final byte[] decompressed = new byte[len];
+
+        Deflater intelDeflater = new IntelDeflaterFactory().makeDeflater(level, nowrap);
+        Inflater javaInflater = new Inflater(nowrap);
+
+        //act
+        intelDeflater.setInput(input,0,input.length);
+        intelDeflater.finish();
+        intelDeflater.deflate(compressed,0,compressed.length);
+        intelDeflater.end();
+
+        javaInflater.setInput(compressed,0,compressed.length);
+        javaInflater.inflate(decompressed,0,decompressed.length);
+        javaInflater.end();
+
+        //assert
+        Assert.assertEquals(decompressed,input);
+    }
+
+    @Test(enabled = true, dataProviderClass = CompressionDataProviders.class, dataProvider = "compressionCases")
+    public void intelDeflationIntelInflationCompatibilityTest(int level, boolean nowrap) throws DataFormatException {
+        //arrange
+        final int len = TestingUtils.SIZE_4MB;
+        final int outputLen = (int) (1.2*TestingUtils.SIZE_4MB);
+
+        final byte[] input = TestingUtils.generateRandomDNAArray(len);
+        final byte[] compressed = new byte[outputLen];
+        final byte[] decompressed = new byte[len];
+
+        Deflater intelDeflater = new IntelDeflaterFactory().makeDeflater(level, nowrap);
+        Inflater intelInflater = new IntelInflaterFactory().makeInflater(nowrap);
+
+        //act
+        intelDeflater.setInput(input,0,input.length);
+        intelDeflater.finish();
+        intelDeflater.deflate(compressed,0,compressed.length);
+        intelDeflater.end();
+
+        intelInflater.setInput(compressed,0,compressed.length);
+        intelInflater.inflate(decompressed,0,decompressed.length);
+        intelInflater.end();
+
+        //assert
+        Assert.assertEquals(decompressed,input);
+    }
+
     @Test(enabled = true)
     public void inputDataTest() {
         final int LEN = 4*1024*1024;
```

Comment:
I see loats of repeatability here, scan we should generate one datset catering to all tests ? This is the same as `SIZE_4MB ` correct?

Reply:
Changed to SIZE_4MB.